harness/determined/cli/token.py

```diff
@@ -0,0 +1,193 @@
+    user = bindings.get_GetUserByUsername(session=sess, username=username).user
+
+    if user is None or user.id is None:
+        raise errors.CliError(f"User '{username}' not found or does not have an ID")
+
+    request = None
+    request = bindings.v1PostAccessTokenRequest(
+        userId=user.id, lifespan=args.expiration_duration, description=args.description
+    )
+    resp = bindings.post_PostAccessToken(sess, body=request).to_json()
+
+    output_string = None
+    if args.yaml:
+        output_string = util.yaml_safe_dump(resp, default_flow_style=False)
+    elif args.json:
+        output_string = json.dumps(resp, indent=2)
+    else:
+        output_string = f'{resp["token"]}\n{resp["tokenId"]}'
+
+    if args.output_file:
+        with open(args.output_file, "w") as file:
+            file.write(output_string)
+    else:
+        print(output_string)
+
+
+def edit_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    try:
+        if args.description and args.token_id:
+            request = bindings.v1PatchAccessTokenRequest(
+                tokenId=args.token_id, description=args.description, setRevoked=False
+            )
+            resp = bindings.patch_PatchAccessToken(sess, body=request, tokenId=args.token_id)
+        print(json.dumps(resp.to_json(), indent=2))
+    except api.errors.APIException as e:
+        raise errors.CliError(f"Caught APIException: {str(e)}")
+    except api.errors.NotFoundException:
+        raise errors.CliError("Token not found")
+    print(f"Successfully updated token with ID: {args.token_id}.")
+
+
+def login_with_token(args: argparse.Namespace) -> None:
+    unauth_session = api.UnauthSession(master=args.master, cert=cli.cert)
+    auth_headers = {"Authorization": f"Bearer {args.token}"}
+    user_data = unauth_session.get("/api/v1/me", headers=auth_headers).json()
+    if "user" in user_data and "username" in user_data.get("user"):
+        username = user_data.get("user").get("username")
+
+    token_store = authentication.TokenStore(args.master)
+    token_store.set_token(username, args.token)
+    token_store.set_active(username)
+    print(f"Authenticated as {username}.")
+
+
+# fmt: off
+
+args_description = [
+    cli.Cmd("to|ken", None, "manage access tokens", [
+        cli.Cmd("describe", describe_token, "describe token info", [
+            cli.Arg("token_id", type=int, nargs=argparse.ONE_OR_MORE, default=None,
+                    help="token id(s) specifying access tokens to describe"),
+            cli.Group(
+                cli.output_format_args["json"],
+                cli.output_format_args["yaml"],
+            ),
+        ]),
+        cli.Cmd("list ls", list_tokens, "list all active access tokens", [
+            cli.Arg("username", type=str, nargs=argparse.OPTIONAL,
+                    help="list token for the given username", default=None),
+            cli.Arg("--only-active", action="store_true", default=None,
```

Comment:
IMO only showing active tokens should be the default behavior, and we should have another flag `--all` to show all revoked/expired ones.

Reply:
Based on our [conversation before](https://hpe-aiatscale.slack.com/archives/C06FU2GV97D/p1727472204228389?thread_ts=1727469784.286809&cid=C06FU2GV97D) we had been running with the idea that `--all` would be returning all tokens for all users, since that's the default behavior when a username isn't specified, we ended up dropping that flag since it was no longer required.
However, your point as to changing the default behavior to only include actives and then enable showing in-actives is a solid idea. And we'll work on implementing it.

This is to mitigate confusion around the `--all` flag because it is used in experiments.
Because that can cause confusion.Because they would have two different behaviors where all for experiments would be listing all experiments across like a workspace outside of just a specific project, and then where and then for us it would be only showing all tokens including revoked and expired.

So instead, if we have something that's explicitly show inactive tokens, then that will include that result.

`det token list <username> --show-inactive`

Reply:
i'm fine with `det token list <username> --show-inactive`! i think my main callout is that yeah, active tokens should be default behavior

Reply:
#NonAdmin:
det token list //active tokens for logged in users
det token list <username> //Error
det token list --show-inactive //all tokens for logged in users

#Admin:
det token list //active tokens for all users
det token list <username> //active tokens for that users
det token list --show-inactive //all tokens for all users
det token list <username> --show-inactive //all tokens for that user

Will add in help text - "list all token the current user has access to"